File: src/main/java/org/academy/api/client/gui/msdf/font/FontStyle.java
package org.academy.api.client.gui.msdf.font;

import org.lwjgl.util.freetype.FreeType;

public enum FontStyle {
    NORMAL,
    BOLD,
    ITALIC,
    BOLD_ITALIC;

    public boolean isBold() {
        return this == BOLD || this == BOLD_ITALIC;
    }

    public boolean isItalic() {
        return this == ITALIC || this == BOLD_ITALIC;
    }

    public static FontStyle of(int styleFlags) {
        if (styleFlags == FreeType.FT_STYLE_FLAG_BOLD) return BOLD;
        if (styleFlags == FreeType.FT_STYLE_FLAG_ITALIC) return ITALIC;
        if (styleFlags == (FreeType.FT_STYLE_FLAG_BOLD | FreeType.FT_STYLE_FLAG_ITALIC)) return BOLD_ITALIC;
        return NORMAL;
    }
}

File: src/main/kotlin/org/academy/api/client/gui/msdf/font/FontStyle.kt
package org.academy.api.client.gui.msdf.font

import org.lwjgl.util.freetype.FreeType

enum class FontStyle {
    NORMAL,
    BOLD,
    ITALIC,
    BOLD_ITALIC;

    val isBold: Boolean
        get() = this == BOLD || this == BOLD_ITALIC

    val isItalic: Boolean
        get() = this == ITALIC || this == BOLD_ITALIC

    companion object {
        fun of(styleFlags: Int): FontStyle {
            return when (styleFlags) {
                FreeType.FT_STYLE_FLAG_BOLD -> BOLD
                FreeType.FT_STYLE_FLAG_ITALIC -> ITALIC
                FreeType.FT_STYLE_FLAG_BOLD or FreeType.FT_STYLE_FLAG_ITALIC -> BOLD_ITALIC
                else -> NORMAL
            }
        }
    }
}